com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ValidateDeployParametersStep.java

```diff
@@ -92,48 +140,59 @@ private void validateDescriptorSize(FileEntry file) {
         }
     }
 
-    private void validateAppArchiveId(DelegateExecution context) {
-        String appArchiveId = StepsUtil.getRequiredString(context, Constants.PARAM_APP_ARCHIVE_ID);
-        String[] appArchivePartsId = appArchiveId.split(",");
-        if (appArchivePartsId.length == 1) {
-            findFile(context, appArchiveId);
-            return;
+    private void validateVersionRule(DelegateExecution context) {
+        String versionRuleString = (String) context.getVariable(Constants.PARAM_VERSION_RULE);
+        try {
+            VersionRule.value(versionRuleString);
+        } catch (IllegalArgumentException e) {
+            throw new SLException(e,
+                                  Messages.ERROR_PARAMETER_1_IS_NOT_VALID_VALID_VALUES_ARE_2,
+                                  versionRuleString,
+                                  Constants.PARAM_VERSION_RULE,
+                                  Arrays.asList(VersionRule.values()));
         }
+    }
 
-        List<FileEntry> archivePartEntries = new ArrayList<>();
-        for (String appArchivePartId : appArchivePartsId) {
-            archivePartEntries.add(findFile(context, appArchivePartId));
+    private void verifyArchiveSignature(DelegateExecution context) {
+        if (!StepsUtil.shouldVerifyArchiveSignature(context)) {
+            return;
         }
+        String archiveFileName = (String) context.getVariable(Constants.VAR_ARCHIVE_FILE_NAME);
+        getStepLogger().debug(Messages.VERIFYING_ARCHIVE_0, archiveFileName);
+        verifyArchiveSignature(archiveFileName);
+        getStepLogger().info(Messages.ARCHIVE_IS_VERIFIED);
+    }
 
-        StepsUtil.setAsJsonBinaries(context, Constants.VAR_FILE_ENTRIES, archivePartEntries);
-        getStepLogger().debug(Messages.BUILDING_ARCHIVE_FROM_PARTS);
-        new ExecutionRetrier().executeWithRetry(() -> new ArchiveMerger(fileService,
-                                                                        getStepLogger(),
-                                                                        context,
-                                                                        logger).createArchiveFromParts(archivePartEntries));
+    private void verifyArchiveSignature(String archiveFileName) {
+        String certificateCN = configuration.getCertificateCN();
+        getStepLogger().debug(Messages.NO_CUSTOM_CERTIFICATE_CN_NAME_USING_DEFAULT_0, certificateCN);
```

Comment:
Change this message to "Will look for certificate CN \"{0}\".". The "using default" part is not relevant anymore, considering that the certificateCN should no longer be configurable through the CLI options.